src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -619,62 +578,53 @@ MavlinkReceiver::handle_message_set_position_target_local_ned(mavlink_message_t
 					pos_sp_triplet.current.valid = true;
 					pos_sp_triplet.current.type = position_setpoint_s::SETPOINT_TYPE_POSITION; //XXX support others
 
-					/* set the local pos values if the setpoint type is 'local pos' and none
-					 * of the local pos fields is set to 'ignore' */
-					if (offboard_control_sp.mode == OFFBOARD_CONTROL_MODE_DIRECT_LOCAL_NED &&
-							!offboard_control_sp_ignore_position_some(offboard_control_sp)) {
+					/* set the local pos values */
+					if (!offboard_control_mode.ignore_position) {
 					pos_sp_triplet.current.position_valid = true;
-					pos_sp_triplet.current.x = offboard_control_sp.position[0];
-					pos_sp_triplet.current.y = offboard_control_sp.position[1];
-					pos_sp_triplet.current.z = offboard_control_sp.position[2];
+					pos_sp_triplet.current.x = set_position_target_local_ned.x;
+					pos_sp_triplet.current.y = set_position_target_local_ned.y;
+					pos_sp_triplet.current.z = set_position_target_local_ned.z;
```

Comment:
Tagging #1788 as this stems from there. This is now a clear drawback of the rebase workflow...